arangod/RestHandler/RestUsersHandler.cpp

```diff
@@ -340,18 +340,43 @@ RestStatus RestUsersHandler::putRequest(auth::UserManager* um) {
   } else if (suffixes.size() == 3 || suffixes.size() == 4) {
     // update database / collection permissions
     std::string const& name = suffixes[0];
+
     if (suffixes[1] == "database") {
       // update a user's permissions
       std::string const& db = suffixes[2];
       std::string coll = suffixes.size() == 4 ? suffixes[3] : "";
+
       if (!isAdminUser()) {
         generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_FORBIDDEN);
+
         return RestStatus::DONE;
       }
 
+      // validate that the collection is present
+      if (suffixes.size() > 3) {
+        auto* databaseFeature = arangodb::application_features::ApplicationServer::lookupFeature<
```

Comment:
Why don't you just use `DatabaseFeature::DATABASE` ??

Reply:
I want to explicitly avoid doing that since public static variables are always a pain for reuse during testing